File: StaticSharp/Modifiers/BoxShadow/BoxShadow.cs
using StaticSharp.Gears;

namespace StaticSharp {

    public interface JBoxShadow : JAbstractBoxShadow {
        public bool Inset { get; set; }
        public double X { get; set; }
        public double Y { get; set; }
        public double Spread { get; set; }
        public double Blur { get; set; }
        public Color Color { get; set; }
    }


    [ConstructorJs]
    public partial class BoxShadow : AbstractBoxShadow {

    }


}

File: StaticSharp/Modifiers/BoxShadow/BoxShadow.js
StaticSharpClass("StaticSharp.BoxShadow", (modifier, element) => {
    StaticSharp.AbstractBoxShadow(modifier, element)
    

    modifier.Reactive = {
        Inset: false,
        X : 0,
        Y: 0,
        Spread: 0,
        Blur: 0,
        Color: () => element.HierarchyForegroundColor,     
    }

    modifier.getBoxShadow = function () {
        let prefix = ""
        if (modifier.Inset) {
            prefix = "inset "
        }
        return prefix + `${modifier.X}px ${modifier.Y}px ${modifier.Blur}px ${modifier.Spread}px ${modifier.Color.toString()}`
    }
})